doc/design/service-discovery.md

```diff
@@ -262,53 +262,45 @@ myListener.attachEvent(caproSubscriber, DATA_RECEIVED, createNotificationCallbac
 #### Event-based notification: Alternative D
 
 ```cpp
-class DiscoveryInfo
+class ServiceDiscovery
```

Comment:
Please remove the implementation from the design document. This is getting extremely fast out of date and absolutely not helpful. A link to the implementation (file path) would be nice.

Reply:
Sorry, but I disagree here. This is not meant to be up to date but provide a higher-level pseudo code, you can think of it as a blueprint or prototype. I think it's good practise the develop a design doc together with some code and tests and later when actually doing the implementation updating the design doc with the latest knowledge gain.

Reply:
But this is not a higher level pseude code. It is a cut and paste from the actual implementation containing even the todos and remarks.